azext_iot/tests/test_iot_ext_unit.py

```diff
@@ -431,6 +453,145 @@ def test_device_update(self, fixture_cmd, serviceclient, req):
+            (generate_device_show_custom(status="disabled"), (generate_device_show_custom(status="enabled"))),
+            (generate_device_show_custom(), (generate_device_show_custom(statusReason="StatusUpdated"))),
+            (
+                generate_device_show_custom(
+                    authentication={
+                        "symmetricKey": {"primaryKey": "", "secondaryKey": ""},
+                        "type": "sas",
+                        "auth_method": "shared_private_key",
+                        "x509Thumbprint": {"primaryThumbprint": "123", "secondaryThumbprint": "321"},
+                    }
+                ),
+                generate_device_show_custom(
+                    authentication={
+                        "symmetricKey": {"primaryKey": "primarykeyUpdated", "secondaryKey": "primarykeyUpdated"},
+                        "type": "sas",
+                        "auth_method": "shared_private_key",
+                        "x509Thumbprint": {"primaryThumbprint": "123", "secondaryThumbprint": "321"},
+                    }
+                )
+            ),
+            (
+                generate_device_show_custom(
+                    authentication={
+                        "symmetricKey": {"primaryKey": "", "secondaryKey": ""},
+                        "type": "selfSigned",
+                        "auth_method": "x509_thumbprint",
+                        "x509Thumbprint": {"primaryThumbprint": "123", "secondaryThumbprint": "321"},
+                    }
+                ),
+                generate_device_show_custom(
+                    authentication={
+                        "symmetricKey": {"primaryKey": "primarykeyUpdated", "secondaryKey": "primarykeyUpdated"},
+                        "type": "selfSigned",
+                        "auth_method": "x509_thumbprint",
+                        "x509Thumbprint": {"primaryThumbprint": "1234", "secondaryThumbprint": "3214"},
+                    }
+                )
+            ),
+            (
+                generate_device_show_custom(
+                    authentication={
+                        "symmetricKey": {"primaryKey": "", "secondaryKey": ""},
+                        "type": "certificateAuthority",
+                        "auth_method": "sas",
+                        "x509Thumbprint": {"primaryThumbprint": "123", "secondaryThumbprint": "321"},
+                    }
+                ),
+                generate_device_show_custom(
+                    authentication={
+                        "symmetricKey": {"primaryKey": "primarykeyUpdated", "secondaryKey": "primarykeyUpdated"},
+                        "type": "certificateAuthority",
+                        "auth_method": "x509_ca",
+                        "x509Thumbprint": {"primaryThumbprint": "123", "secondaryThumbprint": "321"},
+                    }
+                )
```

Comment:
Why are we passing the same function as input

Reply:
Fixed as per our discussion